### PR TITLE
hydrus: Update and improve package

### PR DESCRIPTION
**Pull request by D4ndellion**
###### Motivation for this change

While updating the package I came across many small bugs in the package
I think everything should now be included properly.

I've packaged pyzlma as well since that's a dependency, note that it has some C stuff included in it that I'm not sure what to do with: In https://matrix.to/#/!KqkRjyTEzAGRiZFBYT:nixos.org/$H3u4N1qdSCPZafVmkwQNElZE6BvCOFmw-dwHi_C-ALU?via=nixos.org&via=matrix.org&via=tchncs.de we came to the conclusion to just let it be for now. If you have any comments on it feel free to express them

swftools was marked as broken in #118934, but I'm not sure why. It compiles, but it might be broken for some other reason.
I've tested a couple of the binaries and they seem to work, so marking as unbroken

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131298` at b80d0a6b run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>hydrus</li>
    <li>python38Packages.pylzma</li>
    <li>python39Packages.pylzma</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/duplicate-check-inputs.md">duplicate-check-inputs</a>

Dependencies listed in `propagatedBuildInputs` should not be repeated in `checkInputs`.

Detected duplicates:

- python3.9-nose-1.3.7

Near pkgs/applications/graphics/hydrus/default.nix:53:3:
```
   |
53 |   checkInputs = with pythonPackages; [  mock httmock nose ];
   |   ^
```

Near pkgs/applications/graphics/hydrus/default.nix:27:3:
```
   |
27 |   propagatedBuildInputs = with pythonPackages; [
   |   ^
```
</li>
  </ul>
</details>

**Comment by dali99** [comment]
> Result of `nixpkgs-review pr 131298` at [b80d0a6](https://github.com/NixOS/nixpkgs/commit/b80d0a6b920c5032c8e3f84a39b1d6a95bc26a43) run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 3 packages built successfully:
> 1 suggestion:
> 
>     * warning: [duplicate-check-inputs](https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/duplicate-check-inputs.md)
>       Dependencies listed in `propagatedBuildInputs` should not be repeated in `checkInputs`.
>       Detected duplicates:
>       
>       * python3.9-nose-1.3.7
>       
>       Near pkgs/applications/graphics/hydrus/default.nix:53:3:
>       ```
>          |
>       53 |   checkInputs = with pythonPackages; [  mock httmock nose ];
>          |   ^
>       ```
>       
>       
>           
>             
>           
>       
>             
>           
>       
>           
>         
>       Near pkgs/applications/graphics/hydrus/default.nix:27:3:
>       ```
>          |
>       27 |   propagatedBuildInputs = with pythonPackages; [
>          |   ^
>       ```

removing nose from checkInputs causes the tests to fail

File: pkgs/applications/graphics/hydrus/default.nix [hunk 1]
```diff
@@ -1,23 +1,23 @@
 { lib
 , fetchFromGitHub
-, xz
 , wrapQtAppsHook
 , miniupnpc_2
+, ffmpeg
 , enableSwftools ? false
 , swftools
 , pythonPackages
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/hydrus/default.nix:8]: Can we use python3?

File: pkgs/development/python-modules/pylzma/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  version = "0.5.0";
+
+  # This vendors an old LZMA SDK
+  # After some discussion, it seemed most reasonable to keep it that way
+  # xz, and uefi-firmware-parser also does this
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "074anvhyjgsv2iby2ql1ixfvjgmhnvcwjbdz8gk70xzkzcm1fx5q";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pylzma/default.nix:14]: checkInputs and pythonImportsCheck are missing.

File: pkgs/applications/graphics/hydrus/default.nix (hunk 1 above)
Comment (dali99) [reply pkgs/applications/graphics/hydrus/default.nix:8]: sure

File: pkgs/development/python-modules/pylzma/default.nix (hunk 2 above)
Comment (dali99) [reply pkgs/development/python-modules/pylzma/default.nix:14]: pythonImportsCheck added, there are no checkInputs afaik.

**Pull request merged**